python/pyspark/sql/connect/client.py

```diff
@@ -513,8 +513,11 @@ class SparkConnectClient(object):
     """
 
     @classmethod
-    def retry_exception(cls, e: grpc.RpcError) -> bool:
-        return e.code() == grpc.StatusCode.UNAVAILABLE
+    def retry_exception(cls, e: Exception) -> bool:
+        if isinstance(e, grpc.RpcError):
+            return e.code() == grpc.StatusCode.UNAVAILABLE
+        else:
+            return False
```

Comment:
Hm, why don't we just throw an exception here?

I am sure there are more places than just `SparkSession` that does the connection to the server side, e.g., `DataFrame.collect()`

Reply:
> why don't we just throw an exception here?

The AttemptManager will bubble up the exception when retry_exception is false.

> I am sure there are more places than just SparkSession that does the connection to the server side, e.g., 
DataFrame.collect()

Good point. I will check other places too.